bot/cogs/games/tic_tac_toe.py

```diff
@@ -0,0 +1,296 @@
+            ["", "", ""],
+            ["", "", ""],
+        ]
+
+    def next_player(self) -> None:
+        """Switch player."""
+        if self.player == "x":
+            self.player = "o"
+        else:
+            self.player = "x"
+
+    async def send_board(self) -> None:
+        """Send embed with the board,"""
+        embed = Embed(
+            title="Tic Tac Toe",
+            description=self.str_board,
+            color=Color.blurple()
+        )
+        if not hasattr(self, "msg"):
+            self.msg = await self.ctx.send(embed=embed)
+            await self.add_buttons()
+        else:
+            await self.msg.edit(embed=embed)
+
+    async def add_buttons(self) -> None:
+        """Add reaction buttons to the message."""
+        for reaction in self.reactions:
+            await self.msg.add_reaction(reaction)
+
+    def apply_emoji_move(self, reaction_emoji: str, xo: t.Literal["x", "o"]) -> bool:
+        """Apply move based on an emoji."""
+        try:
+            move_id = self.reactions.index(reaction_emoji)
+        except ValueError:
+            # Bad emoji passed
+            return False
+
+        row = move_id // 3
+        col = move_id % 3
+
+        if not self.board[row][col]:
+            self.board[row][col] = xo
+            return True
+        return False
+
+    @property
+    def str_board(self) -> str:
+        """Stringify the board using emojis."""
+        stringified = ""
+        for row in self.board:
+            stringified += "|".join(self.emojis[col] for col in row)
+            stringified += "\n"
+        return stringified
+
+    def check_win(self) -> t.Tuple[bool, t.Literal["x", "o", None]]:
+        """Check if the board is in win status, if it is return (True, winner)"""
+
+        # Check rows and columns for X streaks
+        row_stat = {"x": [0, 0, 0], "o": [0, 0, 0]}
+        col_stat = {"x": [0, 0, 0], "o": [0, 0, 0]}
+
+        for row_id, row in enumerate(self.board):
+            for col_id, col in enumerate(row):
+                if col:
+                    row_stat[col][row_id] += 1
+                    col_stat[col][col_id] += 1
+
+        if 3 in row_stat["x"] or 3 in col_stat["x"]:
+            return (True, "x")
+        if 3 in row_stat["o"] or 3 in col_stat["o"]:
+            return (True, "o")
+
+        # Check diagonals
+        diag_1 = [row[col] for col, row in enumerate(self.board)]
+        diag_2 = [row[(col + 1) * -1] for col, row in enumerate(self.board)]
+
+        for xo in ["x", "o"]:
+            if all(field == xo for field in diag_1):
+                return (True, xo)
+            if all(field == xo for field in diag_2):
+                return (True, xo)
+
+        return (False, None)
+
+    def check_draw(self) -> bool:
+        """Check if current board is in draw."""
+
+        return all("" not in row for row in self.board)
+
+    async def apply_win(self, xo: t.Literal["x", "o"]) -> None:
+        """If win is True, re"""
```

Comment:
I guess that this docstring isn't complete...

Reply:
Lol, how did I not notice that, thanks